### PR TITLE
start pokenav documentation

### PR DESCRIPTION
**Pull request by ghoulslash**
Start documenting pokenav functions and structs.

## Description
Update pokenav, pokenav_menu_handlers, pokenav_main_menu, match call, condition sub-menu and party/search condition graphs. Documentation is incomplete, but I'm hoping this gets the ball rolling to finish documenting functions/struct elements I couldn't get to or figure out.

The condition graph, mainly functions `InitConditionGraphData` and onwards in `src/menu_specialized.c` were too challenging to try and update. Hopefully someone smarter than me can do it.

## **Discord contact info**
ghoulslash#3839

### CODE REVIEW

File: include/pokenav.h [hunk 1]
```diff
@@ -97,12 +122,12 @@ enum
 	POKENAV_REGION_MAP,
 	POKENAV_CONDITION_PARTY,
 	POKENAV_CONDITION_SEARCH_RESULTS,
-	POKENAV_MENU_9, // Condition
-	POKENAV_MENU_A, // Condition
+	POKENAV_MENU_CONDITION_GRAPH_FROM_SEARCH, // opening condition graph from search list
```
Comment (GriffinRichards) [inline include/pokenav.h:125]: Drop the `MENU` here like the others

File: src/pokenav_conditions_2.c [hunk 2]
```diff
@@ -105,100 +105,100 @@ const struct WindowTemplate gUnknown_0862337C =
...
-    struct Sprite *unk28e0[MAX_CONDITION_SPARKLES];
-    u8 unk2908;
+    struct Sprite *monMarksSprite;
+    struct Sprite *conditionSparkleSprites[MAX_CONDITION_SPARKLES];
+    u8 windowModeState;
     u8 filler2[0x38ac - 0x2909];
 };
 
-extern s8 sub_81CDC84(void); // This function's declaration here is different than its definition in pokenav_unk_6. u8/s8
+extern s8 GetMonMarkIndex(void); // This function's declaration here is different than its definition in pokenav_unk_6. u8/s8
```
Comment (GriffinRichards) [inline src/pokenav_conditions_2.c:142]: Old file name here

File: src/pokenav_menu_handler_2.c [hunk 3]
```diff
@@ -43,54 +43,54 @@ static u32 LoopedTask_ReturnToConditionMenu(s32 state);
...
+static void InitMenuOptionGlow(void);
+static void Task_CurrentMenuOptionGlow(u8 taskId);
+static void SetMenuOptionGlow(void);
+
+static const u16 sPokenavBgDotsPal[] = INCBIN_U16("graphics/pokenav/bg.gbapal");
+static const u32 sPokenavBgDotsTiles[] = INCBIN_U32("graphics/pokenav/bg.4bpp.lz");
+static const u32 sPokenavBgDotsTilemap[] = INCBIN_U32("graphics/pokenav/bg.bin.lz");
+static const u16 sPokenavDeviceBgPal[] = INCBIN_U16("graphics/pokenav/outline.gbapal");
+static const u32 sPokenavDeviceBgTiles[] = INCBIN_U32("graphics/pokenav/outline.4bpp.lz");
+static const u32 sPokenavDeviceBgTilemap[] = INCBIN_U32("graphics/pokenav/outline_map.bin.lz");
```
Comment (GriffinRichards) [inline src/pokenav_menu_handler_2.c:89]: Are the file names for these still accurate? Should be updated if not

File: src/pokenav_conditions_2.c [hunk 4]
```diff
@@ -105,100 +105,100 @@ const struct WindowTemplate gUnknown_0862337C =
...
-    sub_81CE5E4,
-    sub_81CE6BC,
-    sub_81CE700
+    [PARTY_CONDITION_FUNC_NONE] = NULL,
+    [PARTY_CONDITION_FUNC_SLIDE_MON_IN] = LoopedTask_TransitionMons,
+    [PARTY_CONDITION_FUNC_RETURN] = LoopedTask_ExitPartyConditionMenu,
+    [PARTY_CONDITION_FUNC_NO_TRANSITION] = LoopedTask_MoveCursorNoTransition,
+    [PARTY_CONDITION_FUNC_SLIDE_MON_OUT] = LoopedTask_SlideMonOut,
+    [PARTY_CONDITION_FUNC_ADD_MARKINGS] = LoopedTask_OpenMonMarkingsWindow,
+    [PARTY_CONDITION_FUNC_CLOSE_MARKINGS] = LoopedTask_CloseMonMarkingsWindow
```
Comment (GriffinRichards) [inline src/pokenav_conditions_2.c:116]: Align these and other tables same as `sRibbonsMonMenuLoopTaskFuncs`

**Review by GriffinRichards — CHANGES_REQUESTED** [review]
Nice. That's a good deal of pokenav

File: src/pokenav_menu_handler_2.c (hunk 3 above)
Comment (ghoulslash) [reply src/pokenav_menu_handler_2.c]: renamed
`graphics/pokenav/outline.*` to `graphics/pokenav/device_outline.*`
`graphics/pokenav/bg.*` to `graphics/pokenav/bg_dots.*`

**Review by GriffinRichards — APPROVED** [review]

**Pull request merged**